### ISSUE TITLE
Slow query after update

### ISSUE THREAD

**Issue by XuPeng-SH** [duckdb/duckdb#1594]
I’m doing TPC-C benchmark for DuckDB. Duckdb performs well in all cases, except for Payment.  Then I narrowed the problem to two SQL statements:

1. UPDATE CUSTOMER SET C_BALANCE = -3629.64, C_YTD_PAYMENT = 3629.64,   C_PAYMENT_CNT = 2, C_DATA = '899 4 3 4 3 3619.64|wfjqabzipltzcrlmnunfjfnofzvqumcwyjmbwyx  gqzcxjiwsuqdxmunpyqesodlalpckurwpuojuokpsizdzeknbzituuzbsdcybykregspjhecleztajjfydfttijbj  xogrowiajlfuabpnenpkkxzhhlshmzapmzutasbmrkqfssbguprwgfetjfswfsbshgecrmfdwcskorrjudeozbril  nmejinwuqqzgetzbmuaozcmphedcvpavjmlxjebsniaqbkfsfuaaolchxjlrfmefmvhcswfuqijxegugvcjcgrmbm  nvzsxwvogvntbfmwktyloazmndooieuxjhtabmfuyubbbjxemcmpkjvransyumssibfz' WHERE C_W_ID = 3 AN  D C_D_ID = 4 AND C_ID = 899")
2. SELECT C_ID, C_FIRST, C_MIDDLE, C_LAST, C_STREET_1, C_STREET_2, C_CIT  Y, C_STATE, C_ZIP, C_PHONE, C_SINCE, C_CREDIT, C_CREDIT_LIM, C_DISCOUNT, C_BALANCE, C_YTD  _PAYMENT, C_PAYMENT_CNT, C_DATA FROM CUSTOMER WHERE C_W_ID = 4 AND C_D_ID = 7 AND C_LAST   = 'CALLYCALLYPRES' ORDER BY C_FIRST

If the two statements are executed one after the other, the second query takes more than 3 seconds. But if only the second statement is executed, the execution time is only tens of milliseconds.

The data set is the TPC-C data set with warehouse = 10. You can refer to this:
https://github.com/XuPeng-SH/py-tpcc/blob/master/README_DUCKDB.md to mock the data set.
My test program is here: https://github.com/XuPeng-SH/cpp-tpcc

**Comment by Mytherin** [comment]
Thanks for the report!

Could this be down to a checkpoint being triggered by the first query? Is the second query still taking 3 seconds if it is run a second time?

**Comment by XuPeng-SH** [comment]
The second query still takes a long time. I suspect this is caused by some recent code changes. I ran these two SQLs with python API and there was no problem. The python API version is: duckdb==0.2.5.
When I continued to load data to Duckdb, I found that the log file kept increasing, and the data file size remained unchanged, that is, the checkpoint was not triggered until the end. Is this expected? I am worried that it will affect the performance of the query. I expect concurrent write transactions not to affect the performance of AP too much.

BTW, When I use python for benchmarking, the performance is particularly poor. Until I write all statement 
form this style:
con.execute("INSERT INTO items VALUES (?, ?, ?)", ['laptop', 2000, 1])
to:
con.execute(f”INSERT INTO items VALUES (‘{laptop}’, {2000}, {1})”)
